interactive_marker_tutorials/scripts/basic_controls.py

```diff
@@ -131,6 +131,14 @@ def saveMarker( int_marker ):
 #####################################################################
 # Marker Creation
 
+def normalizeQuaternion( quaternion_msg ):
+    norm = quaternion_msg.x**2 + quaternion_msg.y**2 + quaternion_msg.z**2 + quaternion_msg.w**2
+    s = norm**(-0.5)
+    quaternion_msg.x *= s
+    quaternion_msg.y *= s
+    quaternion_msg.z *= s
+    quaternion_msg.w *= s
```

Comment:
As mentioned in the other thread, we should consider providing a function to do this for users.

Ideally in both C++ and Python, perhaps it could live in `sensor_msgs` or similar. There's precedence for this (a message package providing extra API to work with certain messages).

Doesn't need to be done right this second, but it should be used here when it is made available.

Reply:
yep, agreed